### PR TITLE
fix(ui): no hyphen for "create job" action + nice icon (#14776)

### LINKED ISSUE #14776: Create-Job should have an icon and probably not a dash
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

Items in the ⋮ menu generally have icons, but `Create-Job` does not. Also it shouldn't be hyphenated.

**To Reproduce**

1. Find a cron job in argocd
2. Click on ⋮ for it

**Expected behavior**

1. An icon for the action
2. No dash in the action text

**Screenshots**

image

**Version**

```shell
{
    "Version": "v2.8.0+0e198a5.dirty",
    "BuildDate": "2023-06-29T20:40:29Z",
    "GitCommit": "0e198a54a8c47298ca436117c26a9407b4f013ca",
    "GitTreeState": "dirty",
    "GoVersion": "go1.20.5",
    "Compiler": "gc",
    "Platform": "linux/amd64",
    "KustomizeVersion": "v5.1.0 2023-06-19T16:58:18Z",
    "HelmVersion": "v3.12.1+gf32a527",
    "KubectlVersion": "v0.24.2",
    "JsonnetVersion": "v0.20.0"
}
```

**Logs**

```
```

### PR DESCRIPTION
**Pull request by crenshaw-dev**
image

Fixes #14776

### CODE REVIEW

**Review by reggie-k — APPROVED** [review]
Yeah, it looks better with space in the UI. Tests are successful, so looks like no problem in the config as well

**Comment by crenshaw-dev** [comment]
@reggie-k I updated to add an icon and use a new `DisplayName` property instead of renaming the job. Just feels better to avoid spaces in a string that's used programmatically (i.e. in API calls and RBAC evaluation).

**Review by todaywasawesome — APPROVED** [review]

**Comment by jannfis** [comment]
I'm not sure, this looks to introduce a little inconsistency in naming. Other two word actions do have a hyphen, so this would make `create job` quite a unicorn imho.

File: docs/operator-manual/upgrading/2.7-2.8.md [hunk 1]
```diff
@@ -63,7 +63,7 @@ p, role:action-runner, applications, action/apps/Deployment/restart, *, allow
 # If you don't want to grant the new permissions, don't include the following lines
 p, role:action-runner, applications, action/argoproj.io/WorkflowTemplate/create-workflow, *, allow
 p, role:action-runner, applications, action/argoproj.io/CronWorkflow/create-workflow, *, allow
-p, role:action-runner, applications, action/batch/CronJob/create-job, *, allow
+p, role:action-runner, applications, action/batch/CronJob/create job, *, allow
```
Comment (jannfis) [inline docs/operator-manual/upgrading/2.7-2.8.md:66]: Specifically, I mean this one.

Comment (crenshaw-dev) [reply docs/operator-manual/upgrading/2.7-2.8.md:66]: Oops, meant to revert this

Comment (crenshaw-dev) [inline docs/operator-manual/upgrading/2.7-2.8.md:66]:
```suggestion
p, role:action-runner, applications, action/batch/CronJob/create-job, *, allow
```

**Comment by crenshaw-dev** [comment]
@jannfis yep, that's why I eventually went with adding a `displayName` field so we don't create a unicorn.

I just forgot to revert the RBAC line to return to using the hyphen. :-)

**Comment by crenshaw-dev** [comment]
@jannfis fwiw I looked at the other hyphenated actions (I think there's only `promote-full` for Rollouts) and decided that the hyphen makes sense for Promote-Full. "Promote Full" feels awkward.

**Review by jannfis — APPROVED** [review]
LGTM!

**Pull request merged**